v2/tools/generator/internal/codegen/pipeline/convert_allof_and_oneof_to_objects.go

```diff
@@ -160,7 +160,65 @@ func (s synthesizer) getOneOfPropNames(oneOf *astmodel.OneOfType) ([]propertyNam
 		return err
 	})
 
-	return result, err
+	return simplifyPropNames(result), err
+}
+
+// simplifyPropNames makes the OneOf names better by trimming any common suffix
```

Comment:
It trims prefixes (finds common suffixes) right?

Reply:
it finds the common suffixes and then trims them off